app/controllers/spaces.rb

```diff
@@ -0,0 +1,64 @@
+class MakersBnb < Sinatra::Base
+
+  get '/edit/:space_id' do
+    unless current_user
+      redirect '/sessions/new'
+    end
+    @space = Space.get(params[:space_id])
+    if current_user.id == @space.user_id
+      erb :'spaces/edit'
+    else
+      redirect '/users/account'
+    end
+  end
+
+  post '/edit'do
+    Space.get(params[:space_id]).update(name: params[:name], description: params[:description], price: params[:price])
+    redirect to 'users/account'
+  end
+
+  get '/spaces' do
+    if session[:space_array]
```

Comment:
Overusing the `session` here guys - avoid this. There are some nice things going on, but you need to figure an alternative way to store this sort of stuff. `session` is essentially a component of the View, so what you're doing here is pushing stuff as far up the stack as it can go - the opposite of where we should be aiming to push logic.